Added support for access token freshness date.

dunkmann00 opened the pull request:
This pull request adds the ability to pass in a `datetime.timedelta` object for the `fresh` kwarg, instead of a `boolean`, when creating an access token. When using the `fresh_jwt_required` decorator, if the `fresh` value is old, the endpoint will be inaccessible. But the access token is still valid for endpoints that use `jwt_required`.

dunkmann00 commented:
#107

dunkmann00 commented:
Not sure whats going on here. Something related to cookies is failing and I didn't change anything with that. Also, when I clone the master branch and run tox without any changes to the project I also get these errors. Seems strange.

dunkmann00 commented:
Still not sure why this is happening but after further examination it seems that the problem has to do with the `parse_cookie` function which is in `_get_cookie_from_response` function of the `test_cookies.py` file. After passing the cookie into this method it returns a dictionary that leaves off the 'HttpOnly; Path' value.

vimalloc commented:
Looks like changes in the newest werkzeug release: http://werkzeug.pocoo.org/docs/0.14/changes/#version-0-14

I'll get this updated in master soon (today or tomorrow hopefully), then you can rebase this pr and we can get it merged and push out a new release.

Cheers

vimalloc commented:
Ok, assuming all goes well that fix should be in place. If you can rebase this, I would be happy to get it merged in. Cheers 👍

vimalloc commented on tests/test_view_decorators.py:
```diff
@@ -78,6 +78,7 @@ def test_fresh_jwt_required(app):
     with app.test_request_context():
         access_token = create_access_token('username')
         fresh_access_token = create_access_token('username', fresh=True)
+        fresh_timed_access_token = create_access_token('username', fresh=timedelta(minutes=-1))
```
Could we add one more unit test somewhere? You have the not fresh test case here, I would like to have a test added that sets the time delta in the future and successfully accesses a fresh jwt required endpoint while using this new feature. Thanks!

dunkmann00 commented:
Okay so I think I have my branch correct now. This is my first pull request and first time doing a rebase so it took me a little bit but I think its now in good order. I also added the unit test for the successful case and rebased that into my original commit. I hope that was okay to do. Let me know if there is anything else I need to do!

vimalloc commented:
Thanks for working on this!

The rebase against master looks good. I'm not seeing that success unit test in the changes for this pull request though. You can push that into this PR with a separate commit, that doesn't have to be rebased.

Besides that, everything looks great. Once you get that in this PR I'll get this merged.

Thanks for contributing! 👍

dunkmann00 commented:
My mistake! Thought I had that in here already. Just added it now.

Thanks for having started this repo, this is an awesome extension for Flask!
